Scripts/TileMap.gd

```diff
@@ -1,43 +1,84 @@
 extends TileMap
 
-signal proportions(tile_scale, grid_size,tile_size)
+signal proportions(grid_data)
+
+
 #this is the size of each tile
-@export var our_tile_size = Vector2i(16,16)
+@export var our_tile_size = Vector2(16,16)
 
 #how many tiles the grid should be
-@export var our_grid_size = Vector2i(12,12)
+@export var our_grid_size = Vector2(12,12)
+
+#the shape of the grid
+#grid_shape will return selected index
+@export_enum("square","cricle") var grid_shape : int
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
 	create_grid()
 	
-
-
-#creates grid that that scalse with tile size and grid size
+	
+func _process(delta):
+	pass
+	
+	
+#creates grid that that scales with tile size and grid size
 func create_grid():
-	var grid = self
-	var base_tile_size = Vector2i(16,16)
-	var base_grid_size = Vector2i(12,12)
+	var grid_data : Dictionary
+	var selected_grid_shape : String
+	var base_tile_size : Vector2
+	var base_grid_size : Vector2
+	var grid_size_in_units : Vector2
+	var tile_center_offset : Vector2
+	var center_of_grid : Vector2
+	
+	#default grid settings
+	base_tile_size = Vector2(16,16)
+	base_grid_size = Vector2(12,12)
 	
 	
-	grid.our_grid_size = our_grid_size
+	
+	grid_size_in_units = our_tile_size * our_grid_size
+	tile_center_offset = our_tile_size/2
+	center_of_grid = grid_size_in_units/2
 	
 	#needs to be float and vector2(not 2i) otherwise will round
-	grid.scale = Vector2(float(our_tile_size[0])/float(base_tile_size[0]),float(our_tile_size[1])/float(base_tile_size[1]))
-
+	scale = our_tile_size/base_tile_size
```

Comment:
Rather than put this here, you can put it in the variable declaration as: 
`var variable_name := Vector2(x,y)`
Or
`var variable_name := data`

Reply:
I didn't do that for consistency. I don't want some variables to be initialized at the top and some just to be data types. It looks cleaner.

Reply:
The lack of data in an uninitialized variable tells you something. In fact, doing it in one line can help the compiler optimize it. You can separate initialized and uninitialized variables into separate blocks if you like, but keep them separate.

"Looking nice" and "useful readability" are not the same thing.